core/src/test/java/io/parsingdata/metal/IterateTest.java

```diff
@@ -41,11 +42,11 @@
 
     private static final Token repNToken =
         seq(any("n"),
-            repn(def("x", con(1), gtNum(con(1))), ref("n")),
+            repn(def("x", con(1), gtNum(con(1))), last(ref("n"))),
             def("f", con(1), eq(con(42))));
 
     private static final Token repBrokenNToken =
-        seq(repn(any("x"), div(con(1), con(0))),
+        seq(repn(any("x"), last(div(con(1), con(0)))),
```

Comment:
(This looks pretty confusing, to be honest, even though it makes total sense.)

Reply:
Agreed! This is one of those cases where we'd like to have `SingleValueExpression`-_versions_ of common operations such as the logical and math operators.